pkg/client/jobs.go

```diff
@@ -0,0 +1,55 @@
+package client
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *DefaultClient) JobGet(ctx context.Context, name string) (*apiv1.Job, error) {
+	job := &apiv1.Job{}
+	return job, c.Client.Get(ctx, kclient.ObjectKey{
+		Name:      name,
+		Namespace: c.Namespace,
+	}, job)
+}
+
+func (c *DefaultClient) JobList(ctx context.Context, opts *JobListOptions) ([]apiv1.Job, error) {
+	result, listOptions := &apiv1.JobList{}, &kclient.ListOptions{Namespace: c.Namespace}
+
+	if opts != nil && opts.App != "" {
+		listOptions.LabelSelector = k8slabels.SelectorFromSet(map[string]string{labels.AcornAppName: opts.App})
+	}
+
+	if err := c.Client.List(ctx, result, listOptions); err != nil {
+		return nil, err
+	}
+
+	sort.Slice(result.Items, func(i, j int) bool {
+		if result.Items[i].CreationTimestamp.Time == result.Items[j].CreationTimestamp.Time {
+			return result.Items[i].Name < result.Items[j].Name
+		}
+		return result.Items[i].CreationTimestamp.After(result.Items[j].CreationTimestamp.Time)
+	})
+
+	return result.Items, nil
+}
+
+func (c *DefaultClient) JobRestart(ctx context.Context, name string) error {
+	rs := &apiv1.JobRestart{}
+
+	// Since "." is not allowed in k8s job name, we replace it with "-" in the Name
+	// section of the request and send the public name in the labels.
+	rs.Labels = map[string]string{labels.AcornPublicName: name}
```

Comment:
nit, so there is no ambiguity between the object and the request:
```suggestion
	rs := &apiv1.JobRestart{
		ObjectMeta: metav1.ObjectMeta{
			Name: name,
			Namespace: namespace,
			Labels: map[string]string{labels.AcornPublicName: name}
	}

	// Since "." is not allowed in k8s job name, we replace it with "-" in the Name
	// section of the request and send the public name in the labels.
```

Reply:
Second comment: if the `name` doesn't have to be changed to remove the `.`s, do you need this label?

Reply:
This label is no longer needed since I can remove it using the `validator.NoValidation` logic you outlined above.